src/hooks/useBasisForm/validation/validateInput.ts

```diff
@@ -0,0 +1,25 @@
+export const NUMERIC_REGEX = /^\d*$/;
+export const DECIMAL_REGEX = /^\d*(\.\d{2})?$/;
+
+const isInputEmpty = (value: string) => value.trim() === "";
+
+export const validateInput = (
+  value: string,
+  inputProps: Record<string, any>
+) => {
+  const { optional, variant } = inputProps;
+
+  if (isInputEmpty(value) && !optional) {
+    return "Required";
+  }
```

Comment:
When the input is empty, there is no point to continue with other checks.
Let's exit early when input is optional and empty. 
Replace this `if` with:
```
if (isInputEmpty(value)) {
  return optional ? null : "Required".
}
```

Reply:
done